flat_mutation_reader.hh

```diff
@@ -433,6 +433,10 @@ public:
     //
     // When forwarding mode is not enabled, fast_forward_to()
     // cannot be used.
+    //
+    // `partition_start` must be fetched *before* the first call to `fast_forward_to`.
```

Comment:
Also, it must not be called after the last `partition_end`. I would make this statement more general. It can be called only when the reader is within a partition, and it affects that partition's range. A reader is inside a partition between it produces `partition_start` and `partition_end`.

Reply:
That's not exactly true, is it? At some point partition_end was returned when the current ck range was exhausted and then it was ok to call f_f_t again. Unless I remember incorrectly?

Reply:
Or was it just the reader::end_of_stream() returning true?

Reply:
But I agree with the general statement that f_f_t can be called only when the reader is within a partition.

Reply:
But in forwarding mode readers don't return partition_ends

Reply:
@tgrabiec reworded it a bit, check it out

Reply:
Yes. So it's safer to just write within a partition.